### DISCUSSION TITLE
Wagmi signedTypeData hook & metamask sig utils signedTypeData_v4 producing different signature

### DISCUSSION THREAD

**Discussion by liho00** [wevm/wagmi#271] [Q&A]
```
import {
  signTypedData,
} from "@metamask/eth-sig-util";
import {
  useSignTypedData,
} from "wagmi";

const domain = {
  name: "name", // contract deploy name
  version: "1", // contract deploy version
  chainId: 1, // env chain id
  verifyingContract: "",
};

const types = {
  Test: [
    { name: "Request", type: "string" },
  ],
};
const privateKey = Buffer.from(
  "myprivatekey",
  "hex"
);
```
Metamask sig utils SignTypedDataVersion.V4, (tested with SignTypedDataVersion.V3 same)
```
const signature = signTypedData({
  privateKey,
  data: {
    types,
    primaryType: "Test",
    domain,
    message: {
      Request: "This is a request"
    },
  },
  version: SignTypedDataVersion.V4,
});
console.log("Metamask sig utils generated signature", signature); 
```
Metamask sig utils generated signature 0x93f39c57d7f8de7b033367cb5fa7bb68c6a339b0626c73febc57fdce5a779a8472b365951470ade9857e8bd26a3020e9088c110c931abd188034cbdd1e79f8191c

wagmi signTypedData
```
const [_, signTypedData] = useSignTypedData();
signTypedData(domain, types, {
      Request: "This is a request" 
  })
  .then(({data}) => {
    console.log("wagmi signTypedData", signature);
  });
```
wagmi signTypedData 0x9655d8635d4826a1c44185acb34f7c86b286e90e70bd260626c8dd112ae8f5f348f75ce387544c1734971f6ae51ad855e98a49c7e56c3a636a90f714378aca0d1c

My question is why both produce different result of signatures? I was confusing on why these both producing different signature, does it possible both lib produce same signature?

**Comment by 0xPro** [comment]
could you share what you did if you got it solved in the end?

**Comment by phpmac** [comment]
me too
